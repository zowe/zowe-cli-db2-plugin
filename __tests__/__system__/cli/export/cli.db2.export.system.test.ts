/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ITestEnvironment } from "../../../__src__/environment/doc/response/ITestEnvironment";
import { TestEnvironment } from "../../../__src__/environment/TestEnvironment";
import { runCliScript } from "../../../__src__/TestUtils";

let TEST_ENV: ITestEnvironment;

describe("db2 export command", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENV = await TestEnvironment.setUp({
            installPlugin: true,
            tempProfileTypes: ["db2"],
            testName: "export_command"
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENV);
    });

    it("should display the help", () => {
        const response = runCliScript(__dirname + "/__scripts__/export_help.sh", TEST_ENV);
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
    });

    it("should fail with invalid parameter", async () => {
        const response = runCliScript(__dirname + "/__scripts__/invalid_parameter.sh", TEST_ENV);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(1);
    });

    it("should fail with invalid option", async () => {
        const response = runCliScript(__dirname + "/__scripts__/invalid_option.sh", TEST_ENV);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(1);
    });
});
