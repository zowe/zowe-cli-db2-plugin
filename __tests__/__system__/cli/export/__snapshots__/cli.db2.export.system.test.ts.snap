// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`db2 export command should display the help 1`] = `
"
 DESCRIPTION
 -----------

   Export data from a Db2 table

 USAGE
 -----

   zowe db2 export <command>

   Where <command> is one of the following:

 COMMANDS
 --------

   table Export data from a Db2 table in SQL format

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-examples  (boolean)

      Display examples for all the commands in a group

   --help-web  | --hw (boolean)

      Display HTML help in browser

{
  \\"success\\": true,
  \\"exitCode\\": 0,
  \\"message\\": \\"The help was constructed for command: export.\\",
  \\"stdout\\": \\"\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Export data from a Db2 table\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe db2 export <command>\\\\n\\\\n   Where <command> is one of the following:\\\\n\\\\n COMMANDS\\\\n --------\\\\n\\\\n   table Export data from a Db2 table in SQL format\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Export data from a Db2 table\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe db2 export <command>\\\\n\\\\n   Where <command> is one of the following:\\\\n\\\\n COMMANDS\\\\n --------\\\\n\\\\n   table Export data from a Db2 table in SQL format\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n\\"
}"
`;

exports[`db2 export command should fail with invalid option 1`] = `
"Command Error:
Unknown argument: table
Command failed due to improper syntax
Command entered: \\"zowe db2 export --table SYSIBM.SYSDUMMY1\\"
Available commands are \\"table\\".
Use \\"zowe db2 export --help\\" to view groups, commands, and options.
Error Details:
Unknown argument: table
"
`;

exports[`db2 export command should fail with invalid option 2`] = `""`;

exports[`db2 export command should fail with invalid parameter 1`] = `
"Command Error:
Unknown arguments: schema, DEMOUSER.DEMOTABLE
Command failed due to improper syntax
Command entered: \\"zowe db2 export schema DEMOUSER.DEMOTABLE\\"
Available commands are \\"table\\".
Use \\"zowe db2 export --help\\" to view groups, commands, and options.
Error Details:
Unknown arguments: schema, DEMOUSER.DEMOTABLE
"
`;

exports[`db2 export command should fail with invalid parameter 2`] = `""`;
